client/post-editor/post-editor.jsx

```diff
@@ -849,7 +843,7 @@ const PostEditor = React.createClass( {
 
 		// make sure the history entry has the post ID in it, but don't dispatch
 		page.replace(
-			basePath + '/' + this.props.sites.getSite( post.site_ID ).slug + '/' + post.ID,
+			this.props.editPath,
```

Comment:
Huh, I'm really surprised this works, since the logic to update the post ID after saving a new post is only a few lines above here. I would have thought that the prop here wouldn't have been updated until the next tick. Despite this, stepping through in the debugger shows that in-fact the prop is updated in realtime after the call to `this.props.setEditorPostId` . Wonder how that it is?

https://cloudup.com/cRKIECbNlKl